src/gleam/string.gleam

```diff
@@ -739,3 +739,70 @@ pub fn to_option(s: String) -> Option(String) {
     _ -> Some(s)
   }
 }
+
+/// Returns the first element in a grapheme and wraps it in an `Option(String)`.
+/// If the `String` is empty, it returns `None`. Otherwise, it returns
+/// `Some(String)`.
+///
+/// ## Examples
+///
+/// ```gleam
+/// > first("")
+/// None
+/// ```
+///
+/// ```gleam
+/// > first("icecream")
+/// Some("i")
+/// ```
+///
+pub fn first(s: String) -> Option(String) {
+  case length(s) {
+    0 -> None
+    _ -> Some(slice(s, 0, 1))
+  }
+}
+
+/// Returns the last element in a grapheme and wraps it in an `Option(String)`.
+/// If the `String` is empty, it returns `None`. Otherwise, it returns
+/// `Some(String)`.
+///
+/// ## Examples
+///
+/// ```gleam
+/// > last("")
+/// None
+/// ```
+///
+/// ```gleam
+/// > last("icecream")
+/// Some("m")
+/// ```
+///
+pub fn last(s: String) -> Option(String) {
+  case length(s) {
+    0 -> None
+    _ -> Some(slice(s, length(s) - 1, 1))
+  }
+}
+
+/// Creates a new `String` with the first grapheme in the input `String`
+/// converted to uppercase and the remaining graphemes to lowercase.
+///
+/// ## Examples
+///
+/// ```gleam
+/// > capitalize("mamouna")
+/// "Mamouna"
+/// ```
+///
+pub fn capitalize(s: String) -> String {
+  let first =
+    slice(s, 0, 1)
+    |> uppercase
+  let rest =
+    slice(s, 1, length(s))
```

Comment:
Length is quite expensive, I think what you probably want is the pop_grapheme function.